src/core/search/indices.cc

```diff
@@ -20,20 +23,64 @@ using namespace std;
 
 namespace {
 
-// Get all words from text as matched by regex word boundaries
-absl::flat_hash_set<string> TokenizeWords(string_view text) {
-  std::regex rx{"\\b.*?\\b", std::regex_constants::icase};
-  std::cregex_iterator begin{text.data(), text.data() + text.size(), rx}, end{};
+bool IsAllAscii(string_view sv) {
+  return all_of(sv.begin(), sv.end(), [](unsigned char c) { return isascii(c); });
+}
 
-  absl::flat_hash_set<string> words;
-  for (auto it = begin; it != end; ++it) {
-    auto word = it->str();
-    absl::AsciiStrToLower(&word);
-    words.insert(move(word));
+// Get all words from text as matched by the ICU library
+absl::flat_hash_set<std::string> ICUTokenizeWords(std::string_view text) {
+  // Is text contains only ascii, skip working with ICU resources
+  if (IsAllAscii(text)) {
+    std::regex rx{"\\b.*?\\b", std::regex_constants::icase};
+    std::cregex_iterator begin{text.data(), text.data() + text.size(), rx}, end{};
+
+    absl::flat_hash_set<string> words;
+    for (auto it = begin; it != end; ++it) {
```

Comment:
Why not rage-for loop here? 

P.s. you are now on the other side of the mafia  `++it` 🤣

Reply:
1. you can't easily do a for loop if you have two separate iterator -> the for loop construct calls itself begin() and end() functions - on what object in this case?
2. It makes no difference for integers, for iterators I usually pre-increment (not sure if even generic iterators need to have post increment)